tests/analysis/losses/test_losses_base.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+from geopandas import GeoDataFrame
+
+from ra2ce.analysis.analysis_config_wrapper import AnalysisConfigWrapper
+from ra2ce.analysis.analysis_input_wrapper import AnalysisInputWrapper
+from ra2ce.analysis.losses.analysis_losses_protocol import AnalysisLossesProtocol
+from ra2ce.analysis.losses.losses_base import LossesBase
+
+
+class MockLossesSubClass(LossesBase):
+    analysis_input: AnalysisInputWrapper = None
+    analysis: AnalysisLossesProtocol = None
+
+    def __init__(
+        self, analysis_input: AnalysisInputWrapper, config: AnalysisConfigWrapper
+    ):
+        self.analysis_input = analysis_input
+        self.config = config
+
+    def _get_criticality_analysis(self) -> AnalysisLossesProtocol:
+        return MockRedundancyAnalysis(self.analysis_input)
+
+
+class MockRedundancyAnalysis(AnalysisLossesProtocol):
+    analysis_input: AnalysisInputWrapper = None
+
+    def __init__(self, analysis_input: AnalysisInputWrapper):
+        self.analysis_input = analysis_input
+
+    def execute(self) -> GeoDataFrame | None:
+        return GeoDataFrame()
```

Comment:
This "mocking" is not needed as the only thing we are interested to test for the 'base' class it's whether it can be instantiated or not. The remaining tests have been rightfully tested in `test_losses.py`

One reminder on test writing, If the test dictates how the code should behave then there's something wrong going on. This differs from Test / Behavior driven development, where we implement (usually a failing) test based on an agreed functionality that will help us implement the necessary logic so that said tests are successful.

Thus, from my perspective you do not need to test the execution path as it's tightly related to the fact of having a concrete class implementing the `_get_criticality_analysis(self)...`

```suggestion
```

Reply:
Thanks for the explanation